distributed/tests/test_worker.py

```diff
@@ -2837,6 +2845,31 @@ async def test_acquire_replicas_many(c, s, *workers):
         await asyncio.sleep(0.001)
 
 
+@pytest.mark.slow
+@gen_cluster(client=True, Worker=Nanny)
+async def test_acquire_replicas_already_in_flight(c, s, *nannies):
+    """Trying to acquire a replica that is already in flight is a no-op"""
+
+    class SlowToFly:
+        def __getstate__(self):
+            sleep(1.5)
+            return {}
```

Comment:
I think it would be invaluable, in unit tests and stress testing too, to have a tool that simulates slow network comms (for everything and compatible with asyncio). Missing that, this is the best next thing I could think of.

Reply:
Indeed. I built something like this before, see https://github.com/dask/distributed/blob/d0b40d3b9c36e4ebcd61f95baab94c6344e17f1d/distributed/comm/tests/test_comms.py#L883-L903

I've been using a similar approach to "slow deserialization" here https://github.com/dask/distributed/blob/d0b40d3b9c36e4ebcd61f95baab94c6344e17f1d/distributed/tests/test_worker.py#L2615-L2639
I patched the offloading threshold to not have to rely on a nanny. 

I think both slow comm and slow (de-)serialization are valueable tools to have and we should likely properly extract them to a utils function

Lastly, there is also the [`LockedCommPool`](https://github.com/dask/distributed/blob/d0b40d3b9c36e4ebcd61f95baab94c6344e17f1d/distributed/utils_test.py#L1677-L1748) which is a little more difficult to use and very low level but it gives you exact control on when a read/write passes. It also allows you to not have to sleep and makes tests faster

Example usage 
https://github.com/dask/distributed/blob/d0b40d3b9c36e4ebcd61f95baab94c6344e17f1d/distributed/tests/test_utils_test.py#L318-L370

or

https://github.com/dask/distributed/blob/d0b40d3b9c36e4ebcd61f95baab94c6344e17f1d/distributed/tests/test_cancelled_state.py#L85-L104

Reply:
This looks elegant but it doesn't seem to work. I can see the b receiving x from a in microseconds. This I think is because _LockedCommPool only cripples outgoing connections of a, whereas in this case the channel is opened from b to a (b->a ask for x, a->b send back x in the same channel).